torch/csrc/jit/fuser/executor.cpp

```diff
@@ -262,17 +252,22 @@ void launchFusion(
   }
 
   // Adds (flattened) output arguments
+  outputs.reserve(fusion.outputDesc().size());
+  const auto ref_options = device >= 0 ?
+      inputs[0].options().device_index(device) :
+      inputs[0].options().device(at::Device{at::DeviceType::CPU});
   for (size_t i = 0; i < fusion.outputDesc().size(); ++i) {
     const auto& c = fusion.concatDesc()[i];
-    auto& o = outputs[i];
+    const auto& od = fusion.outputDesc()[i];
     if (c.isNoop()) {
-      o.resize_(map_size);
+      outputs.push_back(at::empty(map_size, ref_options.dtype(od.scalar_type)));
```

Comment:
I don’t think we allow having different scalar types at the moment, so this seems unnecessary

Reply:
Don't you think explicitly mentioning is good practice? More future proof code

Reply:
I don’t. It’s not like it will be added next week, so it’s only clutter at the moment. If we ever will want to add this, the person writing th  patch will have to revisit this code anyway.